test_data/rules/prefer_final_parameters.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N prefer_final_parameters`
+
+void badMethod(String label) { // LINT
+  print(label);
+}
+
+void badExpression(int value) => print(value); // LINT
+
+void badMixed(String bad, final String good) { // LINT
+  print(bad);
+  print(good);
+}
+
+bool _testingVariable;
+
+void set badSet(bool setting) => _testingVariable = setting; // LINT
+
+var badCallback = (Object random) { // LINT
```

Comment:
FYI, the spec (and most of the documentation) refers to these as closures, not callbacks.